opal/mca/base/mca_base_alias.c

```diff
@@ -143,10 +143,12 @@ int mca_base_alias_register (const char *project, const char *framework, const c
 
         opal_hash_table_set_value_ptr (alias_hash_table, name, strlen(name), alias);
         free (name);
+        name = NULL;
     }
 
     mca_base_alias_item_t *alias_item = OBJ_NEW(mca_base_alias_item_t);
     if (NULL == alias_item) {
+        if (NULL != name) free(name);
```

Comment:
FWIW, it's safe to `free(NULL)`.  I think we typically are more conservative in the OMPI code base and `if (foo != NULL) free(foo);`, but it's technically not necessary.

Reply:
Yes, I was a bit hesitant to add the checks but then figured I'd stick to the coding style used in throughout the code base,